topdown/http.go

```diff
@@ -841,7 +841,11 @@ func (c *interQueryCache) checkHTTPSendInterQueryCache() (ast.Value, error) {
 
 	headers, err := parseResponseHeaders(cachedRespData.Headers)
 	if err != nil {
-		return nil, err
+		if forceCaching(c.forceCacheParams) {
```

Comment:
`parseResponseHeaders` parses at the etag value as well for instance which is needed in the revalidation server request. So we still need that even if the parsing of the time/date related headers gives an error.

Reply:
Yeah, I'm sort of assuming here that if a server isn't going to support dates it's not going to support etags either. But maybe that's a wrong assumption. What would you suggest here? Should I split up the date and etag processing so that one can succeed without the other? Or should I just make `parseResponseHeaders` less strict so that it doesn't throw and error but just ignores missing or malformed headers, maybe with a log message?

Reply:
I think the only values needed are `etag` and  `last-modified`. Let's see what `parseResponseHeaders` is doing and if the values it's parsing are actually used anywhere.

Reply:
I confirmed that `parseResponseHeaders` is only used with `revalidateCachedResponse` which only looks at `etag`and `last-modified` both of which can be parsed without potential errors. I've made that change to this PR. See what you think.